tasks_downstream.py

```diff
@@ -860,3 +861,102 @@ def after_update(c):
         cur_stat = script_file.stat()
         # Like chmod ug+x
         script_file.chmod(cur_stat.st_mode | stat.S_IXUSR | stat.S_IXGRP)
+
+
+@task(
+    develop,
+    help={
+        "source_db": "The source DB name. Default: 'devel'.",
+        "destination_db": "The destination DB name. Default: '[SOURCE_DB_NAME]-[CURRENT_DATE]'",
+    },
+)
+def snapshot(
+    c,
+    source_db="devel",
+    destination_db=None,
+):
+    """Snapshot current database and filestore.
+
+    Uses click-odoo-copydb behind the scenes to make a snapshot.
+    """
+    if not destination_db:
+        destination_db = "%s-%s" % (
+            source_db,
+            datetime.now().strftime("%Y_%m_%d-%H_%M"),
+        )
+    with c.cd(str(PROJECT_ROOT)):
+        cur_state = c.run("docker-compose stop odoo db", pty=True).stdout
+        _logger.info("Snapshoting current %s DB to %s" % (source_db, destination_db))
+        _run = "docker-compose run --rm -l traefik.enable=false odoo"
+        c.run(
+            f"{_run} click-odoo-copydb {source_db} {destination_db}",
+            env=UID_ENV,
+            pty=True,
+        )
+        if "Stopping" in cur_state:
+            # Restart services if they were previously active
+            c.run("docker-compose start odoo db", pty=True)
+
+
+@task(
+    develop,
+    help={
+        "snapshot_name": "The snapshot name. If not provided,"
+        "the script will try to find the last snapshot"
+        " that starts with the destination_db name",
+        "destination_db": "The destination DB name. Default: 'devel'",
+    },
+)
+def restore_snapshot(
+    c,
+    snapshot_name=None,
+    destination_db="devel",
+):
+    """Restore database and filestore snapshot.
+
+    Uses click-odoo-copydb behind the scenes to restore a DB snapshot.
+    """
+    with c.cd(str(PROJECT_ROOT)):
+        cur_state = c.run("docker-compose stop odoo db", pty=True).stdout
```

Comment:
No need to stop db.

```suggestion
        cur_state = c.run("docker-compose stop odoo", pty=True).stdout
```

Reply:
Well, I actually added this because, without it, I was getting this sort of errors:

```
doodba INFO: Executing click-odoo-copydb devel devel-2021_08_31-09_46
/usr/local/lib/python3.6/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)
2021-08-31 08:46:55,800 1 ERROR ? click_odoo.env_options: exception 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/click_odoo/env_options.py", line 218, in _invoke
    return self.org_invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click_odoo_contrib/copydb.py", line 81, in main
    _copy_db(cr, source, dest)
  File "/usr/local/lib/python3.6/site-packages/click_odoo_contrib/copydb.py", line 25, in _copy_db
    (AsIs(quote_ident(dest, cr)), AsIs(quote_ident(source, cr))),
psycopg2.OperationalError: source database "devel" is being accessed by other users
DETAIL:  There is 1 other session using the database.

Error: source database "devel" is being accessed by other users
DETAIL:  There is 1 other session using the database.
```

So, stopping it ensures all connections are closed before doing anything. Is there a better way of ensuring this?